### PR TITLE
Fix #181: check webhooks status in heartbeat

### LINKED ISSUE #181: Can we ping Bugzilla to see if the webhook is enabled?
We've experienced a few system disruptions caused by Bugzilla disabling the webhook that we've set up to push events to our service.

Is there a way for us to send some request to Bugzilla to check on the status of the webhook? If so, this is something we could potentially incorporate into the healthcheck endpoint.

### PR DESCRIPTION
**Pull request by leplatrem**
Fix #181

* [x] Basic functionnality tested manually on bugzilla-dev.allizom.org
* [x] Unit tests
* [x] Decide what to do with #21 (either clean up disabled hooks or modify heartbeat logics) 
* [x] Wait for deploy of https://bugzilla.mozilla.org/show_bug.cgi?id=1800750 in PROD. See https://bugzilla.mozilla.org/__version__

### CODE REVIEW

**Comment by bsieber-mozilla** [comment]
https://bmo.readthedocs.io/en/latest/extensions/Webhooks/api/v1/webhooks.html

_API + Docs are live in the latest build_

File: jbi/services/bugzilla.py [hunk 1]
```diff
@@ -120,7 +138,28 @@ def get_client():
...
+        for webhook in jbi_webhooks:
+            if not webhook.enabled:
+                all_webhooks_enabled = False
+                logger.error(
+                    "Webhook %s is disabled (%s errors)", webhook.name, webhook.errors
+                )
+
+    health: ServiceHealth = {
+        "up": logged_in,
+        "all_webhooks_enabled": all_webhooks_enabled,
```
Comment (bsieber-mozilla) [inline jbi/services/bugzilla.py:161]: If this is unhealthy; how do we notify about the error? _(sentry? logs? other?)_

**Review by bsieber-mozilla — APPROVED** [review]
Thanks for this PR!

File: jbi/services/bugzilla.py (hunk 1 above)
Comment (bsieber-mozilla) [reply jbi/services/bugzilla.py:161]: I think "errors" can be >0 for a substantial amount of time before the queue is "disabled" but the amount of errors would keep increasing.

I think both "disabled" and "errors" would be worthwhile to track for healthiness

Comment (leplatrem) [reply jbi/services/bugzilla.py:161]: > If this is unhealthy; how do we notify about the error? (sentry? logs? other?)

In this case the `logger.error()` would send a Sentry error.

> I think both "disabled" and "errors" would be worthwhile to track for healthiness

Should we emit a warning when errors are `>0` then?

Comment (bsieber-mozilla) [reply jbi/services/bugzilla.py:161]: > In this case the logger.error() would send a Sentry error.
👍 

> Should we emit a warning when errors are >0 then?
Yeah, a threshold of 0 sounds good.

**Comment by bsieber-mozilla** [comment]
> @bsieber-mozilla Are we already using the `product=Any` webhook? Were all the former webhooks removed? Because if they are still there and disabled the heartbeat will fail. I'm not sure we want to be smart about this, like detect that there is an `Any` webhook enabled, and thus ignore all the disabled ones. Let me know what you think :)

Currently in staging we're using "ANY"; in production we have multiple webhooks that are all enabled. (See #21 and #415.) 

I think we can merge as-is in safely. 

When we make the swap to using "ANY" webhooks, JBI will be unhealthy; it would be good to communicate this transition, since when we do it we will miss events.

_I believe we had received minimal amount of messages between Saturday night and Sunday afternoon; I could make a swap during this time period to reduce the loss of event data._

**Comment by leplatrem** [comment]
> Currently in staging we're using "ANY"; in production we have multiple webhooks that are all enabled. (See https://github.com/mozilla/jira-bugzilla-integration/issues/21 and https://github.com/mozilla/jira-bugzilla-integration/issues/415.)

That means that when this PR is merged, STAGE's heartbeat will fail right? Or have you deleted all the disabled webhooks entries already?

**Comment by bsieber-mozilla** [comment]
> > Currently in staging we're using "ANY"; in production we have multiple webhooks that are all enabled. (See #21 and #415.)
> 
> That means that when this PR is merged, STAGE's heartbeat will fail right? Or have you deleted all the disabled webhooks entries already?

All deleted!
Screenshot 2023-03-06 at 11 39 38 AM

**Pull request merged**